board: arm: Adds mps2 an521 pyocd runner

ithinuel opened the pull request:
This PR enables `PyOCD` as a runner for mps2_an521 [Corstone-201](https://developer.arm.com/Processors/Corstone-201) reference package.

This PR depends on:

- #60780

ithinuel committed:
boards: arm: mps2_an521: add pyocd runner

This allows to run tests & examples on the physical board with:
```
west twister -p mps2_an521 --device-testing --device-serial /dev/ttyUSB0
```

Signed-off-by: Wilfried Chauveau <wilfried.chauveau@arm.com>

ithinuel committed:
test: kernel: mem_protect: switch to an547 as the integration platform

AN521 does not have enough MPU region (8) to handle these tests.

Signed-off-by: Wilfried Chauveau <wilfried.chauveau@arm.com>

nordicjm commented on samples/subsys/ipc/rpmsg_service/Kconfig.sysbuild:
```diff
@@ -0,0 +1,16 @@
+	default "bl5340_dvk_cpunet" if $(BOARD) = "bl5340_dvk_cpuapp"
+	default "lpcxpresso54114_m0" if $(BOARD) = "lpcxpresso54114_m4"
+	default "mps2_an521_remote" if $(BOARD) = "mps2_an521"
+	default "v2m_musca_b1_ns" if $(BOARD) = "v2m_musca_b1"
+	default "v2m_musca_b1_ns" if $(BOARD) = "v2m_musca_b1"
```
duplicate

ithinuel replied:
Oh, indeed, thank you!

nordicjm commented on samples/subsys/ipc/rpmsg_service/sample.yaml:
```diff
@@ -11,6 +11,7 @@ tests:
       - mps2_an521
```
Should the allow list be updated?

ithinuel replied:
What is the allow list in this context?
I still need to re-run this after the holiday break but this test still passed on mps2_an521 provided it was built with sysbuild.

nordicjm commented on samples/subsys/ipc/rpmsg_service/remote/sample.yaml:
```diff
@@ -12,3 +10,4 @@ tests:
     integration_platforms:
       - v2m_musca_b1_ns
```
Is this really needed if both applications are now built using sysbuild?

ithinuel replied:
Maybe, I don't know well enough this part to comment.

nordicjm reviewed (changes requested):
I don't know how CI is passing because this does not work @PerMac ? Before this change I get output like this:
```
Remote core received a message: 88
Master core received a message: 89
Remote core received a message: 90
Master core received a message: 91
Remote core received a message: 92
Master core received a message: 93
Remote core received a message: 94
Master core received a message: 95
Remote core received a message: 96
Master core received a message: 97
Remote core received a message: 98
RPMsg Service demo ended.
Master core received a message: 99
RPMsg Service demo ended.
qemu-system-arm: terminating on signal 2
ninja: build stopped: interrupted by user.
```
I've built with these changes both manually and using twister, twister fails and the log shows the same output, when I run manually I get:
```
qemu-system-arm: warning: nic lan9118.0 has no peer
*** Booting Zephyr OS build zephyr-v3.5.0-2662-g6754bbef3374 ***
Starting application thread!

RPMsg Service [master] demo started
*** Booting Zephyr OS build zephyr-v3.5.0-2662-g6754bbef3374 ***
Starting application thread!

RPMsg Service [remote] demo started
qemu-system-arm: terminating on signal 2
```

So something is very wrong here

PerMac commented:
> Having a look in the logs it hasn't even ran this sample on the integration platform mps2_an521

the sample did passed in the CI. 
```
INFO    -  59/720 mps2_an521                samples/subsys/ipc/rpmsg_service/sample.ipc.rpmsg_service PASSED (qemu 2.388s)
```
at build (1)
I also tested it locally with upstream's docker to confirm

nordicjm commented:
Have re-tested also and it's passing, can't explain why it was failing earlier

carlescufi commented:
@nordicjm can you take another look?

57300 commented on samples/subsys/ipc/rpmsg_service/sysbuild.cmake:
```diff
@@ -0,0 +1,18 @@
+  )
+
+add_dependencies(rpmsg_service rpmsg_service_remote)
+if(("${SB_BOARD}" STREQUAL "lpcxpresso54114_m4"))
+  sysbuild_add_dependencies(CONFIGURE core_m0_inc_target rpmsg_service_remote)
+endif()
```
Can this be removed?
* `SB_BOARD` is undefined, so this code does nothing.
* I don't see `core_m0_inc_target` defined anywhere, neither as a CMake target nor a sysbuild image.

tejlmand reviewed (changes requested):
Seems nrf5340dk_nrf5340_cpuapp is broken when building using sysbuild and this PR.

Will need some time to look further into this.

ithinuel committed:
samples: subsys: ipc: switch rpmsg to sysbuild

rpmsg_service only loads the main application on the target. Switch to
sysbuild which is know to work as expected with the openamp sample.

Signed-off-by: Wilfried Chauveau <wilfried.chauveau@arm.com>

ithinuel commented:
@tejlmand @nordicjm let me know if anything else needs fixing.

nordicjm commented on samples/subsys/ipc/rpmsg_service/remote/CMakeLists.txt:
```diff
@@ -1,25 +1,15 @@
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(rpmsg_service_remote)
 
+message(STATUS "${BOARD} compile as slave in this sample")
```
```suggestion
message(STATUS "${BOARD} compile as remote in this sample")
```

nordicjm reviewed (commented):
Minor nit.

ithinuel committed:
samples: subsys: rpmsg_service improve terminology

Signed-off-by: Wilfried Chauveau <wilfried.chauveau@arm.com>
Co-authored-by: Jamie <40387179+nordicjm@users.noreply.github.com>

tejlmand reviewed (approved):
Approved, the nrf53 issues seems unrelated to this PR.

Thanks for this improvement.
